### PR TITLE
[iterload] added skip keyword to skip first n frames.

### PR DESCRIPTION
**Pull request by marscher**
added a skip keyword to generate time lagged iterators. Currently the NetCDF loader complains about this new keyword, but I dont know how to fix this properly.

### CODE REVIEW

File: mdtraj/tests/test_trajectory.py [hunk 1]
```diff
@@ -478,6 +478,21 @@ def test_iterload():
...
+             'frame0.dcd', 'frame0.binpos', 'legacy_msmbuilder_trj0.lh5',
+             'frame0.xyz', 'frame0.lammpstrj']
+
+    for skip in [1, 2, 10, 99]:
+        for file in files:
+            t_ref = md.load(get_fn(file), top=get_fn('native.pdb'))
+            t = functools.reduce(lambda a, b: a.join(b), md.iterload(get_fn(file), skip=skip, top=get_fn('native.pdb'), chunk=100))
+            eq(t_ref.xyz[skip:], t.xyz)
+            print t_ref.time
+            print t.time
```
Comment (mpharrigan) [inline mdtraj/tests/test_trajectory.py]: python 3 please :)

**Comment by rmcgibbo** [comment]
What is the use case for this?

File: mdtraj/tests/test_trajectory.py (hunk 1 above)
Comment (marscher) [inline mdtraj/tests/test_trajectory.py]: sure, sorry :)

**Comment by marscher** [comment]
* rebased on 1.4.2
* also fixed a bug in topology requiring formats with chunksize=0, where the topology is not being passed, since it is has been popped out of kwargs before the load call.

File: mdtraj/core/trajectory.py [hunk 2]
```diff
@@ -460,29 +464,40 @@ def iterload(filename, chunk=100, **kwargs):
...
         t = load(filename, stride=stride, atom_indices=atom_indices)
         for i in range(0, len(t), chunk):
-            yield  t[i:i+chunk]
+            yield t[i:i+chunk]
 
     else:
         with (lambda x: open(x, n_atoms=topology.n_atoms)
               if extension in ('.crd', '.mdcrd')
               else open(filename))(filename) as f:
+            skipped = 0
```
Comment (rmcgibbo) [inline mdtraj/core/trajectory.py]: it would be more efficient here to just use the `seek` method on the open file handle.

Comment (marscher) [inline mdtraj/core/trajectory.py]: you're so right!

**Pull request merged**